Workiva/dart_build
# AF-308 Initial implementation

# Overview

A CLI that provides dart 1 and dart 2 compatible build and serve commands.

This tool is intended to be used as a transition tool when moving from Dart 1
to Dart 2. Since the `pub serve` and `pub build` commands are not present in
Dart 2, you can use dart_build to run the same command under both versions of
Dart, thus easing the transition to Dart 2. With dart_build you don't need to
modify your pubspec.yaml to add build_runner and build_web_compilers to be
able to build under Dart 2. Those dependencies are included by adding a
dependency on dart_build.

# Changes

  - Copied relevant build_runner pieces from webdev for Dart 2 build and serve
  - Add in dart 1 ability to run pub build & serve
  - Update licensing to include Workiva + original Dart team license

# Testing
 - Clone this repo and checkout this branch
 - Add it as a dev dependency to another project
```
dev_dependencies:
  dart_build:
    path: ../dart_build
```
 - pub get in the other project (with Dart 1)
 - verify that `pub run dart_build serve` runs a server
 - verify that `pub run dart_build build` builds

The dart2 version is on the dart2 branch and updates dependencies for build_runner and SDK constraints to Dart 2. Diff can be viewed here https://github.com/Workiva/dart_build/compare/initial_dart1_functionality...dart2
